Devenv

saimonmoore opened the pull request:
I tried this yesterday. It mostly went quite smoothly. I think it should be merged and then we can continue to iterate on it. Going to add some bits and pieces in another branch and write a bit of documentation.

pmackay committed:
Update nginx config and move to systemd service config

pmackay committed:
Add Google maps key to application.yml template

pmackay committed:
Switch to certbot for letsencrypt setup

pmackay committed:
Add S3 and Stripe vars

enricostano committed:
Ansible configuration to allow tmpfiles reading

enricostano committed:
Add LXC development environment host

enricostano committed:
Remove Vim swap file

enricostano committed:
Fix more stuff

enricostano committed:
Python interpreter

enricostano committed:
Add enricostn pub key

danypr92 committed:
Add dani to users_sysadmin

pmackay committed:
Prevent ansible lint warning

enricostano committed:
WIP

danypr92 committed:
Add conf file and remove arguments

danypr92 committed:
add user to lxc config variables

danypr92 committed:
Set -e flag not works

exist_host variable assignment not works with -e (line 89)

danypr92 committed:
Add provisioning script and cfg file

enricostano committed:
Remove roles not needed in development

enricostano committed:
Add development specific variables

danypr92 committed:
Remove not develoment features

danypr92 committed:
upgrades container script

+ Loop to start cointainer
+ Loop to wait ip

danypr92 committed:
[WIP] provisioning db user

danypr92 committed:
Change db_user name

danypr92 committed:
Uniform db_user between development and production

danypr92 committed:
Remove unused db_superuser

danypr92 committed:
Add default_user and unicorn_user roles as in production

danypr92 committed:
Add mount project to openfoodnetwork user home and refactor ssh-copy-id

danypr92 committed:
[WIP] Create ofn-admin and openfoodnetwork users

- Define default root password
- Change root password
- Change sshd config file to permit root login
- Add argument -e to playbook command to specify the python interpreter
- Remove comments
- Connect ssh whit openfoodnetwork user

danypr92 committed:
Refactor Add host to /etc/hosts and remove old host

danypr92 committed:
Add print info and use vars host and root_passwd

danypr92 committed:
Change openfoodnetwork password and copy ssh key

danypr92 committed:
Change openfoodnetwork for user var

danypr92 committed:
Delete container_user and create app_user. Mount project folder.

- Delete container_user
- Create app_user with uuid 1000
- Mount project folder in app_user home

danypr92 committed:
Add log in create and start container

danypr92 committed:
Move root password change and copy ssh key

danypr92 committed:
Add bundle exec to rake sentences

danypr92 committed:
Use app_user un provisioning script

danypr92 committed:
Copy root ssh key after reboot

danypr92 committed:
Parameterized the roles of db_user in development.yml and provisioning.yml

danypr92 committed:
Add verbose to ansible and refactor db setup

danypr92 committed:
Set -e flag

danypr92 committed:
Ignore files/keys/*

danypr92 committed:
Indent ssh commands

sauloperez commented on inventory/hosts:
```diff
@@ -74,4 +72,3 @@ ie
 in
 uk
 us
-de
```
I'm offended :trollface:

saimonmoore replied:
lol

saimonmoore committed:
Add tmuxinator with default panes

Will start up the rails server and open a console and editor pane

- Automate copying of application.yml.example

saimonmoore committed:
Adding myself to dev playbook users

saimonmoore committed:
Move tmux to new role devloper

sauloperez commented on playbooks/development.yml:
```diff
@@ -44,6 +44,10 @@
       db_user_roles: SUPERUSER,CREATEDB
+
+    - role: devloper # Handy tools for developers
+      tags: devloper
```
is this intended? wouldn't it be better as `developer`?

saimonmoore replied:
it's a pun...@devloper
